repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -1132,7 +1142,7 @@ private Object mapToVertexByTypeCategory(AttributeMutationContext ctx, EntityMut
 
                 AtlasEdge newEdge = mapStructValue(ctx, context);
 
-                if (currentEdge != null && !currentEdge.equals(newEdge)) {
+                if (currentEdge != null && newEdge!= null && !currentEdge.equals(newEdge)) {
```

Comment:
newEdge! is already deleted, so null edge is return  `!currentEdge.equals(null))` will return true and code will try to deleteEdgeReference which is already delete in deleteVertex flow. so `newEdge!= null` condition is required so that it wont attempt to delete already deleted edge